gtda/time_series/embedding.py

```diff
@@ -513,7 +444,238 @@ def resample(self, y, X=None):
         check_is_fitted(self)
         yr = column_or_1d(y)
 
-        yr = np.flip(yr)
-        final_index = -self.time_delay_ * (self.dimension_ - 1)
-        yr = np.flip(yr[:final_index:self.stride])
+        final_index = self.time_delay_ * (self.dimension_ - 1)
+        yr = yr[:final_index - 1:-self.stride][::-1]
         return yr
+
+
+@adapt_fit_transform_docs
+class TakensEmbedding(BaseEstimator, TransformerMixin, PlotterMixin):
+    """Point clouds from collections of time series via independent Takens
+    embeddings.
+
+    This transformer takes collections of (possibly multivariate) time series
+    as input, applies the Takens embedding algorithm described in
+    :class:`SingleTakensEmbedding` to each independently, and returns a
+    corresponding collection of point clouds in Euclidean space (or possibly
+    higher-dimensional structures, see `flatten`).
+
+    Parameters
+    ----------
+    time_delay : int, optional, default: ``1``
+        Time delay between two consecutive values for constructing one embedded
+        point.
+
+    dimension : int, optional, default: ``2``
+        Dimension of the embedding space (per variable, in the multivariate
+        case).
+
+    stride : int, optional, default: ``1``
+        Stride duration between two consecutive embedded points.
+
+    flatten : bool, optional, default: ``True``
+        Only relevant when the input of :meth:`transform` represents a
+        collection of multivariate or tensor-valued time series. If ``True``,
+        ensures that the output is a 3D ndarray or list of 2D arrays. If
+        ``False``, each entry of the input collection leads to an array of
+        dimension one higher than the entry's dimension.
+
+    ensure_last_value : bool, optional, default: ``True``
+        Whether the value(s) representing the last measurement(s) must be
+        be present in the output as the last coordinate(s) of the last
+        embedding vector(s). If ``False``, the first measurement(s) is (are)
+        present as the 0-th coordinate(s) of the 0-th vector(s) instead.
+
+    Examples
+    --------
+    >>> import numpy as np
+    >>> from gtda.time_series import TakensEmbedding
+    # Two univariate time series of duration 4
+    >>> X = np.arange(8).reshape(2, 4)
+    >>> print(X)
+    [[0 1 2 3]
+     [4 5 6 7]]
+    >>> MTE = TakensEmbedding(time_delay=1, dimension=2)
```

Comment:
M like Multi ? xD

Reply:
Hehe, damn, some of them escaped the giant refactor. I'll make a global search for "MTE"...

Reply:
Fixed.